lib/ReactViews/Custom/Chart/ChartExpandAndDownloadButtons.jsx

```diff
@@ -68,6 +72,26 @@ const ChartExpandAndDownloadButtons = createReactClass({
     if (!defined(this.props.sources) && !defined(this.props.tableStructure)) {
       return null;
     }
+
+    // This feels very smelly but it works for the AREMI layers
+    // chartData for a catalogItem returns stuff like
+    // {x: "Butlers Gorge Power Station", y: 67.33}
+    // Occassionally y is null so we know we can't plot it
+    // {x: "Some Station", y: null}
+    // But there isn't a good way to get the chartData for a feature
+    // which is what we ideally need
+    // It doesn't work for SensorObservationService data where chartData is undefined
```

Comment:
Yeah this doesn't seem right to me. Even if the current value is null (which is what we're looking at here I think), there might still be historical data to plot.

Why can't we get the feature's chartData? This component is used alongside a chart on the feature info panel, right? And that chart certainly has the feature's chart data.

Reply:
Because the `chart` and the `chartExpandAndDownloadButtons` are being rendered independently as child components of `FeatureInfoSection` they don't have a shared state of the retrieved chart data I don't think.

I'm wondering if `chartData` ought to be a property, and have a method on the catalogItem called `getChartData`, and then perhaps some sort of `chartDataStatus` property which has a few prescribed values (eg `retrieved`, `retrieving`, `unretrieved` or some way better word).

I think the other challenge in this space is chartData that relates to a feature vs chartData which relates to a whole catalog item. It's a bit unclear in my mind how's that's currently working.

Reply:
Have removed the disabling of buttons for the timebeing, but needs more thought. Have check that nothing breaks at least and that we show the 'no preview available' message properly.